File: src/snippets/shield/reference/remix/DecisionLog.js
import arcjet, { shield, fixedWindow } from "@arcjet/remix";

const aj = arcjet({
  key: process.env.ARCJET_KEY, // Get your site key from https://app.arcjet.com
  rules: [
    fixedWindow({
      mode: "LIVE",
      window: "1h",
      max: 60,
    }),
    shield({
      mode: "LIVE",
    }),
  ],
});

export async function loader(args) {
  const decision = await aj.protect(args);

  for (const result of decision.results) {
    console.log("Rule Result", result);

    if (result.reason.isRateLimit()) {
      console.log("Rate limit rule", result);
    }

    if (result.reason.isShield()) {
      console.log("Shield protection rule", result);
    }
  }

  if (decision.isDenied()) {
    if (decision.reason.isRateLimit()) {
      throw new Response("Too Many Requests", {
        status: 429,
        statusText: "Too Many Requests",
      });
    } else {
      throw new Response("Forbidden", { status: 403, statusText: "Forbidden" });
    }
  }

  return null;
}

File: src/snippets/shield/reference/python-flask/DecisionLog.py
import logging
import os

from arcjet import Mode, arcjet_sync, fixed_window, shield
from flask import Flask, jsonify, request

app = Flask(__name__)

logger = logging.getLogger(__name__)

aj = arcjet_sync(
    key=os.environ["ARCJET_KEY"],  # Get your site key from https://app.arcjet.com
    rules=[
        fixed_window(mode=Mode.LIVE, window=3600, max=60),
        shield(mode=Mode.LIVE),
    ],
)


@app.get("/")
def index():
    decision = aj.protect(request)

    for result in decision.results:
        logger.info("Rule Result %s", result)

        if result.reason_v2.type == "RATE_LIMIT":
            logger.info("Rate limit rule %s", result)

        if result.reason_v2.type == "SHIELD":
            logger.info("Shield rule %s", result)

    if decision.is_denied():
        if decision.reason_v2.type == "RATE_LIMIT":
            return jsonify(error="Too Many Requests"), 429
        return jsonify(error="Forbidden"), 403

    return jsonify(message="Hello world")
